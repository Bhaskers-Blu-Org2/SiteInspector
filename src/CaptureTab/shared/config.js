// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import assignIn from 'lodash.assignin';

const config = {
  defaultTitle: 'Capture',
  id: 'capture',
  stateNamespace: 'CaptureTabState',
  captureDelay: 5000,
  siteInspectorElement: 'site-inspector',
};

// This is used to pass parameters to the tab from the parent config object.
export const setParameters = (overrideConfig) => {
  if (overrideConfig && Object.keys(overrideConfig).length) {
    assignIn(config, overrideConfig);
  }
};

export default config;
