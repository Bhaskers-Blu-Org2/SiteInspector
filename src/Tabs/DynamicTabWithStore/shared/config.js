// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import assignIn from 'lodash.assignin';

const config = {
  defaultTitle: 'DynamicTab',
  id: 'dynamictab',
  stateNamespace: 'DynamicTabState',
};

// This is used to pass parameters to the tab from the parent config object.
export const setParameters = (overrideConfig) => {
  if (overrideConfig && Object.keys(overrideConfig).length) {
    assignIn(config, overrideConfig);
  }
};

export default config;
