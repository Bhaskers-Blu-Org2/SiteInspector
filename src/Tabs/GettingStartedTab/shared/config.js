// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import assignIn from 'lodash.assignin';

const config = {
  defaultTitle: 'My First Tab',
  id: 'gettingstartedtab',
  stateNamespace: 'CustomTabState',
};

// This is used to pass parameters to the tab from the parent config object.
export const setParameters = (overrideConfig) => {
  if (overrideConfig && Object.keys(overrideConfig).length) {
    assignIn(config, overrideConfig);
  }
};

export default config;
