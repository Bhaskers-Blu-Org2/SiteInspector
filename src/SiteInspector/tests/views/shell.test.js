// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import React from 'react';
import configureStore from 'redux-mock-store';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ShellContainer from '../../views/Shell/container';
import config from '../../shared/config';

describe('Primary container view tests', () => {
  const initialState = {
    ShellState: {
      currentPath: '',
      position: 'right',
      tabs: [],
      visible: false,
      routesUpdated: false,
    },
    SharedState: {
      siteInspectorInitialized: true,
    },
  };
  const mockStore = configureStore();
  let store;
  let container;
  let shell;

  configure({ adapter: new Adapter() });

  beforeEach(() => {
    // suppress react import warning
    console.error = jest.fn();

    config.title = 'SI Test';

    store = mockStore(initialState);
    container = shallow(<ShellContainer store={store} />);
  });

  it('Container was created', () => {
    expect(container.length).toEqual(1);
  });

  it('SiteInspector rendered properly', () => {
    expect(container).toMatchSnapshot();
  });

  describe('Shell component tests', () => {
    beforeEach(() => {
      store = mockStore(initialState);
      container = shallow(<ShellContainer store={store} />);
      shell = container.dive();
    });

    it('Shell is created', () => {
      expect(shell.length).toEqual(1);
    });

    it('Tab is rendered', () => {
      expect(shell.find('.shell-client-tab').length).toEqual(1);
    });

    it('Panel is rendered', () => {
      const panel = shell.find('.panel');
      expect(panel.length).toEqual(1);
      expect(panel.hasClass('visible')).toBeFalsy();
      expect(panel).toMatchSnapshot();
    });

    it('Panel visbility is toggled after click', () => {
      const panel = shell.find('.panel');
      shell.find('.shell-client-tab').simulate('click');
      setTimeout(() => {
        expect(panel.hasClass('visible')).toBeTruthy();
        shell.find('.shell-client-tab').simulate('click');
        setTimeout(() => {
          expect(panel.hasClass('visible')).toBeFalsy();
        }, 1500);
      }, 1500);
    });

    it('Panel has router', () => {
      const panel = shell.find('.panel');
      expect(panel.find('ShellInspectorRouter').length).toEqual(1);
    });

    it('Panel has menu', () => {
      const panel = shell.find('.panel');
      expect(panel.find('.si-header-right').length).toEqual(1);
      const menu = panel.find('.si-header-right');
      expect(menu.find('ActionMenu').length).toEqual(1);
    });

    it('Panel has correct title', () => {
      const panel = shell.find('.panel');
      const title = panel.find('.si-header-title');
      expect(title.length).toEqual(1);
      expect(title.text()).toEqual('SI Test');
    });
  });
});
