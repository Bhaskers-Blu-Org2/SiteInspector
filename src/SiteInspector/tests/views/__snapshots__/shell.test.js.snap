// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary container view tests Shell component tests Panel is rendered 1`] = `
<div
  className="panel flex-column  right"
>
  <div
    className="header"
    id="siHeader"
  >
    <div
      className="si-header-left"
    >
      <a
        href="https://github.com/Microsoft/SiteInspector/blob/master/README.md"
        rel="noopener noreferrer"
        target="_blank"
      >
        <[object Object]
          className="panel-icon"
        />
      </a>
      <div
        className="si-header-title"
      >
        SI Test
      </div>
    </div>
    <div
      className="si-header-right"
    >
      <ActionMenu
        actionTrigger={
          <span
            className="si-glyph si-glyph-settings"
          />
        }
        menu={
          Array [
            Object {
              "content": "Position Left",
              "id": "siMenuPosition",
              "onClick": [Function],
            },
            Object {
              "content": "About",
              "id": "siMenuAbout",
              "onClick": [Function],
            },
          ]
        }
      />
      <div
        className="si-panel-glyph"
        title="Close StoreInspector"
      >
        <span
          aria-hidden="true"
          className="si-glyph si-glyph-cancel"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="panel-content-wrapper flex-column"
  >
    <ShellInspectorRouter
      initialPath=""
      resetRouteUpdate={[Function]}
      routesUpdated={false}
      setPath={[Function]}
      tabs={Array []}
    />
  </div>
</div>
`;

exports[`Primary container view tests SiteInspector rendered properly 1`] = `
<Shell
  currentPath=""
  onToggleShell={[Function]}
  position="right"
  removeMessage={[Function]}
  resetRouteUpdate={[Function]}
  routesUpdated={false}
  setCurrentPath={[Function]}
  setShellPanelPosition={[Function]}
  siteInspectorInitialized={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  tabs={Array []}
  visible={false}
/>
`;
